go/vt/vtgate/planbuilder/route.go

```diff
@@ -622,13 +622,41 @@ func (rb *route) computePlan(pb *primitiveBuilder, filter sqlparser.Expr) (opcod
 			return rb.computeINPlan(pb, node)
 		case sqlparser.NotInOp:
 			return rb.computeNotInPlan(node.Right), nil, nil
+		case sqlparser.LikeOp:
+			return rb.computeLikePlan(pb, node)
 		}
 	case *sqlparser.IsExpr:
 		return rb.computeISPlan(pb, node)
 	}
 	return engine.SelectScatter, nil, nil
 }
 
+// computeLikePlan computes the plan for 'LIKE' constraint
+func (rb *route) computeLikePlan(pb *primitiveBuilder, comparison *sqlparser.ComparisonExpr) (opcode engine.RouteOpcode, vindex vindexes.SingleColumn, condition sqlparser.Expr) {
+
+	left := comparison.Left
+	right := comparison.Right
+
+	if sqlparser.IsNull(right) {
+		return engine.SelectNone, nil, nil
+	}
+
+	vindex = pb.st.Vindex(left, rb)
+	if vindex == nil {
+		// if there is no vindex defined, scatter
+		return engine.SelectScatter, nil, nil
+	}
+
+	if !rb.exprIsValue(right) {
+		return engine.SelectScatter, nil, nil
+	}
+	if subsharding, ok := vindex.(*vindexes.CFC); ok {
```

Comment:
This is not great. WDYT of making this part of the Vindex interface, where you can ask the vindex if it supports sub-sharding, instead of doing type casts like this?

That way, custom vindex implementations can reproduce this behaviour

Reply:
good point. Added `vindexes.Prefixable`